src/test/rules-engine/index-test.js

```diff
@@ -536,23 +540,23 @@ describe('rules engine', function() {
     expect(res.body).toHaveProperty('id');
     const ruleId = res.body.id;
 
-    await sleep(200);
-
-    Events.add(new Event('surge',
-                         'oh no there is too much electricity',
-                         thingLight1.id));
-
-    await sleep(200);
-
-    res = await chai.request(server)
-      .get(`${Constants.THINGS_PATH}/${thingLight1.id
-      }${Constants.ACTIONS_PATH}`)
-      .set('Accept', 'application/json')
-      .set(...headerAuth(jwt));
-    expect(res.status).toEqual(200);
-    expect(Array.isArray(res.body)).toBeTruthy();
-    expect(res.body.length).toEqual(1);
-    expect(res.body[0]).toHaveProperty('blink');
+    await waitForExpect(async () => {
+      Events.add(new Event('surge',
```

Comment:
The adding the event should be outside of the waitForExpect so that the test only adds one event.

Reply:
Thanks for the review. Since responding the rule-engin API dose not means that rule can work, I change that the rule-engin waits response until connecting websocket.